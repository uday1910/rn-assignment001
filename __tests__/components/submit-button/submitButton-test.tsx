import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import SubmitButton from '../../../source/components/submit-button/submitButton';

const mockPressEvent = jest.fn();

const containerStyle = {
  alignItems: 'center',
  justifyContent: 'center',
};

const props = {
  onPressButton: mockPressEvent,
  style: containerStyle,
  buttonTitle: 'Button',
};

describe('SubmitButton Function', () => {
  it('SubmitButton render properly according to the snapshot', () => {
    const button = renderer.create(<SubmitButton {...props} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('should have a button', async () => {
    const {getByTestId} = render(<SubmitButton {...props} />);
    const button = await getByTestId('submit-button');
    expect(button).toBeDefined();
  });

  it('able to press button', () => {
    const {getByTestId} = render(<SubmitButton {...props} />);
    fireEvent.press(getByTestId('submit-button'));
    expect(mockPressEvent).toBeCalled();
  });
});
