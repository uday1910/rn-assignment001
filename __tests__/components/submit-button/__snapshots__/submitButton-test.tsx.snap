// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton Function SubmitButton render properly according to the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "85%",
    }
  }
  testID="submit-button"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#3498DB",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;
